pkg/permissionclaim/permissionclaim_labeler.go

```diff
@@ -111,3 +115,29 @@ func (l *Labeler) LabelsFor(ctx context.Context, cluster logicalcluster.Name, gr
 
 	return labels, nil
 }
+
+func isSelected(claim apisv1alpha1.PermissionClaim, name, namespace string) bool {
+	// All and ResourceSelector are mutually exclusive. Validation should catch this, but don't leak info if it doesn't somehow.
+	if claim.All && len(claim.ResourceSelector) > 0 {
+		return false
+	}
+
+	// ResourceSelector nil check to be compatible with objects created prior to the addition of the All field
+	if claim.All || len(claim.ResourceSelector) == 0 {
+		return true
+	}
+
+	for _, selector := range claim.ResourceSelector {
+		// Selecting a specific object, might be cluster-scoped.
+		if selector.Name == name && (selector.Namespace == namespace || selector.Namespace == "") {
```

Comment:
nit: for a cluster-scoped object, `selector.Namespace == namespace` should still eval to true, right? since you'd need to not set the ns on selection?

Reply:
I _think_ so; I'm honestly trying to remember why I did this check. I think that might have been with objects incoming via the mutating admission plugin.

Reply:
I re-evaluated this line, and I now remember what I was trying to accomplish.

`selector.Namespace == ""` matches cluster-scoped resources, *or* the resource's name in any namespace.

However, given this ambiguity, perhaps we should have a constant for `*` that indicates any namespace and leave `""` to mean cluster-scoped. Thoughts? @sttts @s-urbaniak

Reply:
+1 to some distinguisher. A thought that come to mind:

Make `namespaces` a list. If the resource being claimed is cluster-scoped, don't set this field. If it's namespace-scoped, we could either (1) require at least 1 item, which could be `*` to mean all namespaces, or (2) have no value and/or an empty list mean all namespaces.

WDYT?

Reply:
I think the list idea makes sense, as it allows for more flexibility.

I also like `len(nsList) == 0` as the cluster-scoped along with `*` for all, as that gives us a way to clearly distinguish user intent as opposed to the current state.